app/app_kmac.c

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2022, Cisco Systems, Inc.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://github.com/cisco/libacvp/LICENSE
+ */
+
+
+#include <openssl/evp.h>
+#include <openssl/param_build.h>
+#include "acvp/acvp.h"
+#include "app_lcl.h"
+#include "safe_lib.h"
+
+#ifdef ACVP_NO_RUNTIME
+# include "app_fips_lcl.h"
+#endif
+
+int app_kmac_handler(ACVP_TEST_CASE *test_case) {
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+    ACVP_KMAC_TC *tc;
+    EVP_MAC *mac = NULL;
+    EVP_MAC_CTX *kmac_ctx = NULL;
+    OSSL_PARAM_BLD *pbld = NULL;
+    OSSL_PARAM *params = NULL;
+    const char *alg_name = NULL;
+    unsigned char *mac_compare = NULL;
+    int rv = 1, diff = 1;
+    size_t mac_out_len;
+    ACVP_SUB_KMAC alg;
+
+    if (!test_case) {
+        return rv;
+    }
+
+    tc = test_case->tc.kmac;
+    if (!tc) return rv;
+    if (!tc->key) return rv;
+
+    alg = acvp_get_kmac_alg(tc->cipher);
+    if (alg == 0) {
+        printf("Invalid cipher value in KMAC");
+        return 1;
+    }
+
+    switch (alg) {
+    case ACVP_SUB_KMAC_128:
+        alg_name = "KMAC-128";
+        break;
+    case ACVP_SUB_KMAC_256:
+        alg_name = "KMAC-256";
+        break;
+    default:
+        printf("Error: Unsupported KMAC algorithm requested by ACVP server\n");
+        return rv;
+    }
+
+    mac = EVP_MAC_fetch(NULL, alg_name, NULL);
+    if (!mac) {
+        printf("Error: unable to fetch KMAC");
+        goto end;
+    }
+    kmac_ctx = EVP_MAC_CTX_new(mac);
+    if (!kmac_ctx) {
+        printf("Error: unable to create KMAC CTX");
+        goto end;
+    }
+
+    pbld = OSSL_PARAM_BLD_new();
+    if (!pbld) {
+        printf("error creating param_bld in KMAC\n");
+        goto end;
+    }
+    OSSL_PARAM_BLD_push_utf8_string(pbld, "cipher", alg_name, 0);
+  // OSSL_PARAM_BLD_push_octet_string(pbld, "key", tc->key, tc->key_len);
```

Comment:
Just using it directly below, so this is not needed ?

Reply:
Yep, forgot to take that line out. Ill take care of in next PR. Thanks!